src/store/request-password-reset/saga.ts

```diff
@@ -0,0 +1,59 @@
+import { requestPasswordReset as requestPasswordResetApi } from './api';
+import { setLoading, setStage, setErrors, SagaActionTypes, RequestPasswordResetStage } from './index';
+import { ResetPasswordErrors } from '.';
+import { call, put, take, fork, cancel } from 'redux-saga/effects';
+
+export function validateRequestPasswordResetEmail({ email }) {
+  const validationErrors = [];
+  if (!email || !email.trim()) {
+    validationErrors.push(ResetPasswordErrors.EMAIL_REQUIRED);
+  }
+  return validationErrors;
+}
+
+export function* requestPasswordReset() {
+  while (true) {
+    const action = yield take(SagaActionTypes.RequestPasswordReset);
+    yield put(setLoading(true));
+
+    const validationErrors = validateRequestPasswordResetEmail(action.payload);
+
+    if (validationErrors.length > 0) {
+      yield put(setErrors(validationErrors));
+      continue;
+    }
+
+    try {
+      const result = yield call(requestPasswordResetApi, action.payload);
+
+      if (result.success) {
+        yield put(setStage(RequestPasswordResetStage.Done));
+        break;
+      } else {
+        let errors = [];
+        if (result.response === 'UNKNOWN_ERROR') {
+          errors.push(ResetPasswordErrors.UNKNOWN_ERROR);
+        }
+        yield put(setErrors(errors));
```

Comment:
This seems off. What if the response sends a different type of error....then we'd end up with an empty error list.

```suggestion
        yield put(setErrors([ResetPasswordErrors.UNKNOWN_ERROR]));
```